organization/models.py

```diff
@@ -2,105 +2,145 @@
-    # Styret
-    ("styret", "Styret"),
-    # Økonomi
-    ("okonomi", "Økonomi"),
-)
-
 
 class InternalGroup(models.Model):
-    """
-    An internal group within KSG, e.g. Lyche bar
-    """
-    STATUS = KSG_INTERNAL_GROUPS
-
-    name = StatusField(unique=True)
-
-    description = models.CharField(max_length=1024, blank=True, null=True)
+    class Meta:
+        default_related_name = 'groups'
+        verbose_name_plural = 'Internal groups'
 
-    members = models.ManyToManyField(
-        User,
-        blank=True,
-        related_name='internal_groups'  # The default django user model already has a `groups` related_name
-        # so we have to make a custom one
-    )
+    class Type(models.TextChoices):
+        """
+        Making use of https://docs.djangoproject.com/en/3.1/ref/models/fields/#enumeration-types
+        Denotes the internal group type, either a interest-group, e.g. KSG-iT, Påfyll etc. or an internal
+        grouping, e.g. Lyche bar og Edgar
+        """
+        INTEREST_GROUP = "interest-group"
+        INTERNAL_GROUP = "internal-group"
+
+    name = models.CharField(unique=True, max_length=32)
+    type = models.CharField(max_length=32, null=False, blank=False, choices=Type.choices)
+    description = models.TextField(max_length=2048, blank=True, null=True)
+
+    @property
+    def slugify(self):  # is this stupid way to do it? Just use ID's instead?
+        return "-".join(self.name.lower().split(" ")).replace("ø", "o").replace("æ", "ae").replace("å", "a")
+
+    @property  # consider this not being a property seeing as its more "computationally" expensive than normal properties?
+    def active_members(self):
+        """
+        Returns all group positions membership with a FK to positions with a FK to this instance. Returns
+        only "active" memberships which is defined as memberships without an ending date
+        """
+        group_members = []
+        for position in self.positions.all():
+            group_members.extend(position.active_memberships.all())
+
+        group_members.sort(key=lambda x: x.user.get_full_name())
+        return group_members
 
     def __str__(self):
         return "Group %s" % self.name
 
     def __repr__(self):
         return "Group(name=%s)" % self.name
 
-    class Meta:
-        default_related_name = 'groups'
-        verbose_name_plural = 'Internal groups'
+
+class InternalGroupPositionMembership(models.Model):
+    """
+    An intermediary model between a user and a InternalGroupPosition with additional information
+    regarding membership
+    """
+    date_joined = models.DateField(auto_now_add=True)
+    date_ended = models.DateField(
+        default=None,
+        null=True,
+        blank=True
+    )
+
+    user = models.ForeignKey("users.User", on_delete=models.CASCADE)
+    position = models.ForeignKey("organization.InternalGroupPosition", related_name="memberships",
+                                 on_delete=models.DO_NOTHING)
```

Comment:
Is it more correct with CASCADE here? I reckon that a internal group should never be deleted as long as there are members. Maybe CASCADE here and DO_NOTHING on the position model?

Reply:
I feel like `CASCADE` is the way to go here? When a position is deleted, clearly the membership to that position should be deleted, yes?